pkg/cloud/alibaba/assets/cloud-controller-manager-deployment.yaml

```diff
@@ -0,0 +1,108 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  labels:
+    app: alibaba-cloud-controller-manager
+  name: alibaba-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: alibaba-cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        app: alibaba-cloud-controller-manager
+    spec:
+      hostNetwork: true
+      serviceAccountName: cloud-controller-manager
+      priorityClassName: system-cluster-critical
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: "kubernetes.io/hostname"
+              labelSelector:
+                matchLabels:
+                  app: alibaba-cloud-controller-manager
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+          operator: Exists
+        - effect: NoExecute
+          key: node.kubernetes.io/unreachable
+          operator: Exists
+          tolerationSeconds: 120
+        - effect: NoExecute
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+          tolerationSeconds: 120
+        - effect: NoSchedule
+          key: node.cloudprovider.kubernetes.io/uninitialized
+          operator: Exists
+        - effect: NoSchedule
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+      containers:
+        - command:
+            - /bin/sh
+            - -c
+            - |
+              #!/bin/sh
+              set -o allexport
+              if [[ -f /etc/kubernetes/apiserver-url.env ]]; then
+                source /etc/kubernetes/apiserver-url.env
+              fi
+              exec /cloud-controller-manager \
+              --kubeconfig=/etc/kubernetes/cloud-controller-manager.conf \
+              --address=127.0.0.1 \
+              --allow-untagged-cloud=true \
+              --leader-elect=true \
```

Comment:
There are more leader election flags required to be configured, the unit tests should explain the issues (you can also copy/paste the leader election flags from any other platform)